File: next-js-boilerplate/src/api/client/rtc/calls-actions.ts
import type { RtcReportReason } from "@/api/server/rtc/shared-types";

export function useCallActions() {
  const reportCall = async (
    callId: string,
    reason: RtcReportReason,
    details?: string,
  ) => {
    const { reportCallServer } = await import("@/api/server/rtc/calls/report");
    return reportCallServer(callId, reason, details);
  };

  return { reportCall };
}

File: flutter-boilerplate/lib/api/client/rtc/calls_actions.dart
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../server/rtc/calls_report.dart';

final callActionsProvider = Provider((ref) => CallActions(ref));

class CallActions {
  final Ref _ref;

  CallActions(this._ref);

  Future<void> report(String callId, String reason, {String? details}) => _ref
      .read(reportCallServerProvider)
      .call(callId, reason, details: details);
}
